src/ripple/app/tx/impl/AMMDeposit.h

```diff
@@ -0,0 +1,214 @@
+ * their relative price, then no trading fee is charged on the transaction.
+ * The trader can specify different combination of the fields in the deposit.
+ * LPTokens - transaction assumes proportional deposit of pools assets in
+ *     exchange for the specified amount of LPTokens of the AMM instance.
+ * Asset1In - transaction assumes single asset deposit of the amount of asset
+ *     specified by Asset1In. This is essentially an equal asset deposit
+ *     and a swap.
+ * Asset1In and Asset2In - transaction assumes proportional deposit of pool
+ *     assets with the constraints on the maximum amount of each asset that
+ *     the trader is willing to deposit.
+ * Asset1In and LPTokens - transaction assumes that a single asset asset1
+ *     is deposited to obtain some share of the AMM instance's pools
+ *     represented by amount of LPTokens.
+ * Asset1In and EPrice - transaction assumes single asset deposit with
+ *     the following two constraints:
+ *         a. amount of asset1 if specified in Asset1In specifies the maximum
+ *             amount of asset1 that the trader is willing to deposit
+ *         b. The effective-price of the LPTokens traded out does not exceed
+ *             the specified EPrice.
+ * Following updates after a successful AMMDeposit transaction:
+ * The deposited asset, if XRP, is transferred from the account that initiated
+ *     the transaction to the AMM instance account, thus changing the Balance
+ *     field of each account.
+ * The deposited asset, if tokens, are balanced between the AMM account
+ *     and the issuer account trustline.
+ * The LPTokens are issued by the AMM instance account to the account
+ *     that initiated the transaction and a new trustline is created,
+ *     if there does not exist one.
+ * The pool composition is updated.
+ */
+class AMMDeposit : public Transactor
+{
+public:
+    static constexpr ConsequencesFactoryType ConsequencesFactory{Custom};
+
+    explicit AMMDeposit(ApplyContext& ctx) : Transactor(ctx)
+    {
+    }
+
+    static TxConsequences
+    makeTxConsequences(PreflightContext const& ctx);
+
+    /** Enforce constraints beyond those of the Transactor base class. */
+    static NotTEC
+    preflight(PreflightContext const& ctx);
+
+    /** Enforce constraints beyond those of the Transactor base class. */
+    static TER
+    preclaim(PreclaimContext const& ctx);
+
+    /** Gather information beyond what the Transactor base class gathers. */
+    void
+    preCompute() override;
+
+    /** Attempt to create the AMM instance. */
+    TER
+    doApply() override;
+
+private:
+    std::pair<TER, bool>
+    applyGuts(Sandbox& view);
+
+    /** Deposit requested assets and token amount into LP account.
+     * @param view
+     * @param ammAccount AMM account
+     * @param asset1Deposit deposit amount
+     * @param asset2Deposit deposit amount
+     * @param lpTokensDeposit amount of tokens to deposit
+     * @return
+     */
```

Comment:
The comments here don't add value - it's just repeating the param names. Ditto for other functions here. I'd probably just leave off params that aren't documented. I.e. `ammAccount AMM account` doesn't really add anything.